agent/spec/spec_helper.rb

```diff
@@ -74,6 +74,11 @@
     end
   end
 
+  config.around :each do |ex|
+    Timeout.timeout(5.0) do
```

Comment:
So this is a safeguard that broken specs don't just hang indefinitely, but fail with an error. If the specs just hang, then it's very hard to tell from e.g. the travis logs which spec got stuck.